### PR TITLE
Commit

### PR DESCRIPTION
**Pull request by mensfeld**
@agwozdowski  @pavlo-vavruk  please review - it is a fix for #63 
also protects from kill -9 and loosing data then and other weird cases that might occur.
Significantly speeds up shutdown time (from 20-60 seconds on a full load to less than 1 second)

### CODE REVIEW

File: lib/karafka/cli.rb [hunk 1]
```diff
@@ -47,7 +46,7 @@ def cli_commands
   # Reloads Karafka irb console session
   def reload!
     puts "Reloading...\n"
-    Kernel.exit 10
+    Kernel.exec "KARAFKA_CONSOLE=true bundle exec irb -r #{Karafka.boot_file}"
```
Comment (pavlo-vavruk) [inline lib/karafka/cli.rb]: Can we use here ` Kernel.exec  Karafka::Cli::Console.command` ?

File: lib/karafka/connection/queue_consumer.rb [hunk 2]
```diff
@@ -81,6 +83,27 @@ def target
...
+      # @param partition [Integer] Number of partition from which we were consuming messages
+      # @param last_processed_message [Poseidon::FetchedMessage] last message that we've
+      #   successfully processed
+      # @note By processed we mean passing it through the controller and scheduling it to be
+      #   handled by the worker. Then we can consider data to be "saved" since it is being stored
+      #   in Redis and will be handled in a background job
+      # @note Most of the time, this method will be scheduled on a last message that we've received
+      #   from Kafka, but there are some cases (graceful shutdown, restart) when we will stop
+      #   processing fetched messages, return last that we've managed to handle and exit
+      # @note If is also worth pointing out, that if we kill Karafka process during the data
```
Comment (pavlo-vavruk) [inline lib/karafka/connection/queue_consumer.rb]: `@note If`  => `@note It`

**Comment by pavlo-vavruk** [comment]
@mensfeld reviewed

File: lib/karafka/cli.rb (hunk 1 above)
Comment (mensfeld) [inline lib/karafka/cli.rb]: yeah good catch - that was a purpose of that!

**Comment by mensfeld** [comment]
@pavlo-vavruk  thanks! Added tweaks and once the CI is green, will merge

**Pull request merged**